### DISCUSSION TITLE
The compatibility issues regarding the upgrade from 6.x.

### DISCUSSION THREAD

**Discussion by love720720** [apache/skywalking#12159] [Q&A]
Currently, our company is using version 6.x of SkyWalking and we're considering upgrading to version 9.x. According to the documentation, compatibility with previous versions ceased starting from version 8.x. However, due to the multitude of projects in our production environment reliant on version 6.x, it's not feasible to transition directly to version 9.x all at once. Additionally, in our practical application process, we heavily depend on traceId, as it helps us pinpoint certain issues in the logs (using log4j2). We would like to inquire about how to ensure compatibility with traceId from version 6.x in version 9.x. At the same time, we also need to ensure compatibility with traceId from version 9.x in version 6.x. The essential requirement is that the completion of the chain can be reflected in the logs.

Thanks

**Comment by wu-sheng** [comment]
Single answer is, there is no compatibility support.

And 6.x is 4 years ago, so, don't expect anything from community.
For us, v10 is coming next month.

**Reply by wu-sheng** [reply]
If you use Java agent, the trace ID is not changed about format and log4j integration. This part is good.

**Reply by love720720** [reply]
I didn't find a relevant compatibility solution indeed, but still, thank you for your response. Thanks.

**Reply by wu-sheng** [reply]
There isn't. v8 breaks a lot to improve.
